ocaml/xapi/xapi_pool.ml

```diff
@@ -1077,6 +1077,54 @@ let designate_new_master ~__context ~host =
     Xapi_pool_transition.attempt_two_phase_commit_of_new_master ~__context true peers my_address
   end
 
+let management_reconfigure ~__context ~network =
+  (* Disallow if HA is enabled *)
+  let pool = Helpers.get_pool ~__context in
+  if Db.Pool.get_ha_enabled ~__context ~self:pool then
+    raise (Api_errors.Server_error(Api_errors.ha_is_enabled, []));
+
+  (* Create a hash table for hosts with pifs for the network *)
+  let pifs_on_network = Db.Network.get_PIFs ~__context ~self:network in
+  let hosts_with_pifs = Hashtbl.create 16 in
+  List.iter (fun self ->
+    let host = Db.PIF.get_host ~__context ~self in
+    Hashtbl.add hosts_with_pifs host self;
+  ) pifs_on_network;
+
+  (* All Hosts must have associated PIF on the network *)
+  let all_hosts = Db.Host.get_all ~__context in
+  List.iter (fun host ->
+    if not(Hashtbl.mem hosts_with_pifs host) then
+      raise (Api_errors.Server_error(Api_errors.pif_not_present, [Ref.string_of host; Ref.string_of network]));
+  ) all_hosts;
+
+  (* Constraints on the PIFs:
+   * PIFs must not be bond slaves.
+   * PIFs must have same primary_address_type
+   * PIFs must not have ip_configuration_mode as None *)
+  let address_type = Db.PIF.get_primary_address_type ~__context ~self:(List.hd pifs_on_network) in 
+  List.iter (fun self ->
```

Comment:
As mentioned earlier, please replace this with
```
List.iter (fun self ->
  Xapi_pif.assert_usable_for_management ~__context self;
  if Db.PIF.get_primary_address_type ~__context ~self <> address_type then
    raise (Api_errors.Server_error(Api_errors.pif_incompatible_primary_address_type, [Ref.string_of self]));
) pifs_on_network;
```
and call `Xapi_pif.assert_usable_for_management` from the appropriate place in `Xapi_host.management_reconfigure` as well (instead of using individual calls to `assert_valid_ip(v6?)_configuration` and `assert_not_bond_slave`).

Doing it this way means less code in total (good in itself), but the reason I'm so insistent on this is that if in future we need to change the set of checks, then we will be able to change just one function, `Xapi_pif.assert_usable_for_management`, and this will update both the Pool and Host functions: we avoid the risk of the Pool and Host functions using different checks.